simulatingrisk/risky_food/model.py

```diff
@@ -0,0 +1,141 @@
+from enum import Enum
+from statistics import mean
+
+import mesa
+
+
+class FoodChoice(Enum):
+    RISKY = "R"
+    SAFE = "S"
+
+
+class FoodStatus(Enum):
+    CONTAMINATED = "C"
+    NOTCONTAMINATED = "N"
+
+
+class Agent(mesa.Agent):
+    def __init__(self, unique_id, model, risk_level=None):
+        super().__init__(unique_id, model)
+        # get a random risk tolerance; returns a value between 0.0 and 1.0
+        self.risk_level = risk_level or self.random.random()
+
+    def step(self):
+        # choose food based on the probability not contaminated and risk tolerance
+        if self.risk_level > self.model.prob_notcontaminated:
+            choice = FoodChoice.RISKY
+        else:
+            choice = FoodChoice.SAFE
+        self.payoff = self.model.payoff(choice)
+
+
+class RiskyFoodModel(mesa.Model):
+    prob_notcontaminated = None
+
+    def __init__(self, n):
+        self.num_agents = n
+        self.schedule = mesa.time.SimultaneousActivation(self)
+        # initialize agents for the first round
+        for i in range(self.num_agents):
+            a = Agent(i, self)
+            self.schedule.add(a)
+
+        self.nextid = i + 1
+
+        self.datacollector = mesa.DataCollector(
+            model_reporters={
+                "prob_notcontaminated": "prob_notcontaminated",
+                "contaminated": "contaminated",
+                "average_risk_level": "avg_risk_level",
+                "min_risk_level": "min_risk_level",
+                "max_risk_level": "max_risk_level",
+                "num_agents": "total_agents",
+            },
+            agent_reporters={"risk_level": "risk_level", "payoff": "payoff"},
+        )
```

Comment:
Interesting setup they provid

Reply:
seems very verbose / redundant to me, but at least workable enough for now